src/transactions/kolour/kolour-nft.ts

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {
+  Lucid,
+  UTxO,
+  UnixTime,
+  C,
+  Core,
+  fromHex,
+  Address,
+  Lovelace,
+} from "lucid-cardano";
+
+import { getPaymentKeyHash } from "@/helpers/lucid";
+import * as S from "@/schema";
+import { KolourNftMintingRedeemer } from "@/schema/teiki/kolour";
+import { Hex } from "@/types";
+import { assert } from "@/utils";
+
+import { KOLOUR_TX_MAX_DURATION } from "./constants";
+
+export type Kolour = string; // RRGGBB
+
+export type KolourListing = {
+  fee: Lovelace;
+  image: string; // ipfs://<cid>
+};
+
+export type Quotation = {
+  kolours: Record<Kolour, KolourListing>;
+  userAddress: Address;
+  feeAddress: Address;
+  referral?: string;
+  expiration: number; // Unix Timestamp in seconds
+};
+
+export type Params = {
+  quotation: Quotation;
+  kolourNftRefScriptUtxo: UTxO;
+  txTime: UnixTime;
```

Comment:
Pass both txTimeStart and txTimeEnd, since we might use something lower than `10min` for the TTL.
```suggestion
  txTimeStart: UnixTime;
  txTimeEnd: UnixTime;
```